python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,504 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+import lsst.pex.exceptions as pexExcept
+
+__all__ = ["OverscanCorrectionTaskConfig", "OverscanCorrectionTask"]
+
+
+class OverscanCorrectionTaskConfig(pexConfig.Config):
+    """Overscan correction options.
+    """
+    fitType = pexConfig.ChoiceField(
+        dtype=str,
+        doc="The method for fitting the overscan bias level.",
+        default='MEDIAN',
+        allowed={
+            "POLY": "Fit ordinary polynomial to the longest axis of the overscan region",
+            "CHEB": "Fit Chebyshev polynomial to the longest axis of the overscan region",
+            "LEG": "Fit Legendre polynomial to the longest axis of the overscan region",
+            "NATURAL_SPLINE": "Fit natural spline to the longest axis of the overscan region",
+            "CUBIC_SPLINE": "Fit cubic spline to the longest axis of the overscan region",
+            "AKIMA_SPLINE": "Fit Akima spline to the longest axis of the overscan region",
+            "MEAN": "Correct using the mean of the overscan region",
+            "MEANCLIP": "Correct using a clipped mean of the overscan region",
+            "MEDIAN": "Correct using the median of the overscan region",
+            "MEDIAN_PER_ROW": "Correct using the median per row of the overscan region",
```

Comment:
Add `MEANCLIP_PER_ROW`?

Reply:
Not on this ticket.  I think after the next latiss run there will be more work on sorting out the peculiar overscan structure that prompted the MEDIAN_PER_ROW (which in turn prompted this rewrite so new methods can be added easier in the future).